### PR TITLE
Fix the approval count of PR when there is no protection branch rule

### PR DESCRIPTION
**Pull request by nanguanlin6**
As title
ksnip_20230926-115158

### CODE REVIEW

**Comment by silverwind** [comment]
Fixes https://github.com/go-gitea/gitea/issues/27208?

**Comment by lng2020** [comment]
> Fixes #27208?

I don't think so. The `offical` field involves many repo settings and I'm not sure about the root reason.

**Comment by lng2020** [comment]
related(#27286)

File: models/issues/review.go [hunk 1]
```diff
@@ -242,14 +243,21 @@ func IsOfficialReviewer(ctx context.Context, issue *Issue, reviewers ...*user_mo
 		return false, err
 	}
 	if rule == nil {
-		return false, nil
+		// if no rule is found, then user with write access can make official reviews
+		repo, err := repo_model.GetRepositoryByID(ctx, pr.BaseRepoID)
```
Comment (lunny) [inline models/issues/review.go]: Use pr.LoadBaseRepo because it maybe loaded.

**Review by lunny — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Pull request merged**